File: starter/src/main/java/com/udacity/store/model/Order.java
package com.udacity.store.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(name = "orders")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Order {
//TODO: Add the details of the order class

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @OneToMany
    private List<Product> products;

    @JoinColumn(name = "total")
    private Double total;
    @JoinColumn(name = "customer_name")
    private String customerName;
    @JoinColumn(name = "customer_address")
    private String customerAddress;
    @JoinColumn(name = "customer_credit_card")
    private String customerCreditCard;

}

File: starter/src/main/webapp/angular-frontend/src/app/models/Order.ts
import { Product } from "./Product";

//TODO: Add the contents of the order class
export class Order{
    id:number;
    name:string;
    price:number;
    products:Product[];
    // totalOrder: number;
    

    constructor(){
        this.id = 0;
        this.name ="";
        this.price = 0; 
        this.products=[];
        // this.totalOrder=0;
    }

}